pkg/kubelet/util/util.go

```diff
@@ -60,3 +65,50 @@ func GetContainerByIndex(containers []v1.Container, statuses []v1.ContainerStatu
 	}
 	return containers[idx], true
 }
+
+var (
+	tmpfsNoswapOptionSupported        bool
+	tmpfsNoswapOptionAvailabilityOnce sync.Once
+)
+
+const TmpfsNoswapOption = "noswap"
+
+func IsTmpfsNoswapOptionSupported(mounter mount.Interface) bool {
+	isTmpfsNoswapOptionSupportedHelper := func() bool {
+		if sysruntime.GOOS == "windows" {
+			return false
+		}
+
+		mountDir, err := os.MkdirTemp("", "tmpfs-noswap-test")
```

Comment:
Good point.
@kwilczynski maybe you can help answering that?

Reply:
There are two things to address here:

- Does Kubernetes have a known, perhaps even dedicated, location somewhere on the file system it can use?
- Would there always be access to the common temporary directory for Kubernetes?

The answer to the first question would be: I am unaware of such a directory or a mount point that Kubernetes creates and uses exclusively for its own needs. Usually, the role of such location somewhere on the file system would have been filed by the common temporary directory. However, perhaps Kubernetes should provide this under, let's say, `/run/kubelet`? or `/run/kubernetes`? or perhaps `/run/k8s`? This is where temporary files, directories, tests of sorts, tag files, etc., could be stored. This would also make it easier to make SELinux or AppArmor policies, and such, should there be a need for such a Kubernetes-curated location.

Second question: Using a common system-wide shared temporary directory is something that, on most Linux systems, would have been taken for granted and provided. Any process, whether running as a privileged user or not, should be able to save some temporary data it needs on the side there. We have /tmp and /var/tmp as being the two most commonly used locations, and a lot of libraries tend to hard code at least one of these locations (/tmp most commonly since it's ephemeral in terms of how its content lifespan looks like). Even with systemd and private temporary directory support enabled, a process should have no issues writing to its assumed default location.

Now, whether Kubernetes could be restricted from using /tmp or /var/tmp, of course it could, as it would be something that the user has done on purpose, most of the time, since by default, the expectation is that writes to files or directories under /tmp will succeed.

Having said all that, the default temporary directory location would be a perfectly reasonable choice for Kubernetes. Should it fail, then we are dealing with something non-standard on the underlying node, or within some runtime environment, which would be something different to look into and the user will be most likely aware of that, hopefully.

See the following for more details about temporary directories:

- [Using /tmp/ and /var/tmp/ Safely](https://systemd.io/TEMPORARY_DIRECTORIES)
- [file-hierarchy — File system hierarchy overview](https://www.freedesktop.org/software/systemd/man/latest/file-hierarchy.html)

Reply:
> * Does Kubernetes have a known, perhaps even dedicated, location somewhere on the file system it can use?
> * Would there always be access to the common temporary directory for Kubernetes?
> 
> The answer to the first question would be: I am unaware of such a directory or a mount point that Kubernetes creates and uses exclusively for its own needs.

isn't that the kubelet RootDirectory?

https://github.com/kubernetes/kubernetes/blob/7a6931b31cf04d2a91ce8c0af09ad0f474927436/pkg/kubelet/kubelet.go#L1380-L1430

Reply:
@liggitt, that would work. Given the following code comment:

```go
// getRootDir returns the full path to the directory under which kubelet can
// store data.  These functions are useful to pass interfaces to other modules
// that may need to know where to write data without getting a whole kubelet
// instance.
func (kl *Kubelet) getRootDir() string {
	return kl.rootDirectory
}
```

Can we use it to write anything? Or are there some expectations of the type of data being written there? If not, then it would work fine indeed.

@iholder101, thoughts?

Reply:
Thanks a lot for this discussion!
This is implemented in [`e8b303d` (#124060)](https://github.com/kubernetes/kubernetes/pull/124060/commits/e8b303d5da6d20a6628bbd7e35eebfe3fe2cd658).